src/context/node.js

```diff
@@ -318,6 +339,55 @@ export class ContextNode {
     }
   }
 
+  /**
+   * @param {string} name
+   * @param {function(!Node):boolean} match
+   * @param {number} weight
+   * @return {!ContextNode}
+   */
+  addGroup(name, match, weight) {
+    const groups = this.groups || (this.groups = new Map());
+    const {node, children} = this;
+    const cn = new ContextNode(node, name);
+    groups.set(name, {cn, match, weight});
+    cn.setParent(this);
+    if (children) {
+      children.forEach(discoverContextNode);
+    }
+    return cn;
+  }
+
+  /**
+   * @param {string} name
+   * @return {?ContextNode}
+   */
+  group(name) {
+    const {groups} = this;
+    const group = groups && groups.get(name);
+    return (group && group.cn) || null;
+  }
+
+  /**
+   * @param {!Node} node
+   * @return {?ContextNode}
+   * @protected
+   */
+  matchGroup(node) {
```

Comment:
nit: maybe a name like `computeActiveGroup`? I'm used to a `match` func returning bool

Reply:
Renamed to `findGroup`